server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -119,6 +119,7 @@ public class RemoteClusterStateService implements Closeable {
     private volatile TimeValue slowWriteLoggingThreshold;
 
     private final AtomicBoolean deleteStaleMetadataRunning = new AtomicBoolean(false);
+    private final RemoteClusterStateStats remoteStateStats;
```

Comment:
can this tracking be done in RemotePersistedState ?

Reply:
It'll require passing listeners in writeIncrementalMetadata and deleteClusterUUIDs method to capture cleanup attempt failed count. That can be done but looks like RemoteClusterStateService can cleanly keep the remoteStateStats.

I'll try to provide it via RemotePersistedState only as that can access RemoteClusterStateService.